pkg/controller/controller_instance_test.go

```diff
@@ -1136,6 +1225,133 @@ func TestPollServiceInstanceSuccessDeprovisioningWithOperationWithFinalizer(t *t
 	assertNumEvents(t, events, 1)
 }
 
+func TestSetInstanceCondition(t *testing.T) {
```

Comment:
more comments in this test. It is too difficult to read quickly without studying all the parts.